### DISCUSSION TITLE
Encrypting/decrypting large files

### DISCUSSION THREAD

**Discussion by codedust** [panva/jose#558] [Q&A]
Hi! Does anyone here have experience with encrypting/decrypting large files using the JOSE framework? Most implementations use `strings` or `Uint8Array`s, which means that all plaintext and ciphertext must be kept in memory during encryption/decryption.

I was thinking about using the [Streams API](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API) to allow encryption without keeping all the plaintext and ciphertext in memory.

Has anyone used JOSE for large files and have some tips?

**Comment by panva** [comment]
There is no streaming in this module at this moment. The underlying Web APIs don't support streaming anyway.

**Reply by codedust** [reply]
Implementing streaming encryption seems possible even with the Web Crypto API: https://github.com/hwcwy/AES-GCM-stream/ (disclaimer: I did not check the implementation for correctness).

(This is not a feature request. I'm just looking for experiences on this topic - also in the non-JS world if someone around here has some tips.)
